### DISCUSSION TITLE
Any plans for a date picker?

### DISCUSSION THREAD

**Discussion by amreus** [charmbracelet/gum#409] [Ideas]
A date picker would be quite useful.

I don't think I really need to expand too much on it. Dates are obviously useful to humans.

A couple modes come to mind tho. 

Mode 1: A calendar mode where the current month is displayed

        August 2023
    Su Mo Tu We Th Fr Sa
           1  2  3  4  5
     6  7  8  9 10 11 12
    13 14 15 16 17 18 19
    20 21 22 23 24 25 26
    27 28 29 30 31

and the cursor keys can be used to selected a day. 

Mode 2: an inline mode where the date is shown as

    Select a date: 2023-08-12

and the some shortcut keys can increment/decrement day or month or year.

That's it. Thanks.

**Comment by lukasschwab** [comment]
I threw together the "inline mode" on a fork because I needed it: https://github.com/lukasschwab/gum/pull/1

Happy to clean it up and open a PR, but I realize it's likely to clash with https://github.com/charmbracelet/bubbles/pull/76.

**Reply by lukasschwab** [reply]
@maaslalani I'm happy to open a gum PR while the [bubbles picker](https://github.com/charmbracelet/bubbles/pull/475) is a work in progress; seems stalled since March.

A component in bubbles seems clearly preferable; if https://github.com/charmbracelet/bubbles/pull/475 is merged, it should replace what I hacked together. In the mean time, a simple date-picker — just the compact behavior here — is a useful thing.

**Reply by meowgorithm** [reply]
Hi! We should _probably_ look to merge https://github.com/charmbracelet/bubbles/pull/475, however can you provide some sample code we can use to vet your fork?

**Reply by lukasschwab** [reply]
Sure — feel free to specify a format you want!

+ I added some lines to `test.sh`: https://github.com/lukasschwab/gum/pull/1/files#diff-2107b9bb7be9ac26a632056369875d18f7f6091852d889612def0acdc7e54cf0
+ I added a vhs demo to my README: ![demo of `gum date`](https://camo.githubusercontent.com/580dfc3e499bbdbce57063c9edbe3c144a9081513d9821482334f6398e11ad98/68747470733a2f2f7668732e636861726d2e73682f7668732d32476b69656d7830414c5a5a426d4f446353726732492e676966)

Usage is pretty straightforward:

> Pick a date, starting from the current date by default:
> 
> ```bash
> gum date
> ```
> 
> Or starting from an initial date of your choosing:
> 
> ```bash
> gum date --value 2023-11-28
> ```

The fork may need some TLC before it's ready for a PR — a rebase, reverting some superfluous changes and perhaps some proper unit testing. Here's the diff: https://github.com/lukasschwab/gum/commit/969153b2c4a8e75598646ed6870602d5e850eadd

**Reply by lukasschwab** [reply]
I'll clean up and open a PR tonight, just so you have something concrete to reference.

For what it's worth, I'd almost always prefer a compact picker like this — maybe with some added press-enter-to-type-in-field functionality — over a calendar picker. Most of my use cases require picking a date at some stable offset from a default date.

**Reply by lukasschwab** [reply]
PR open: https://github.com/charmbracelet/gum/pull/683
